docs/API.md

```diff
@@ -153,8 +153,7 @@ Basic data type: string
 
 This parameter is a string that are an IPv4 or IPv6. It's validated with the following regexes:
  - IPv4 : `/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/`
- - IPv6 : `/^([0-9A-Fa-f]{1,4}:[0-9A-Fa-f:]{1,}(:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})?)|([0-9A-Fa-f]{1,4}::[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})$/`
-
+ - IPv6 : `/^[0-9a-f:]+(:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})?$/i`
```

Comment:
This is not fully correct. 300.300.300.300 matches the expression for IPv4, but it is not valid. I think it would be better to write:
```
IPv4: valid IPv4 address in [dot-decimal notation]
IPv6: valid IPv6 address in the [recommend text format for IPv6 addresses]


[Dot-decimal notation]: https://en.wikipedia.org/wiki/Dot-decimal_notation
[Recommend text format for IPv6 addresses]: https://datatracker.ietf.org/doc/html/rfc5952

Reply:
Two things that are really nice about regular expressions is that they're portable and easy to compare. This is useful when you want to review the correctness of the implementation or when you want to perform client-side validation that's identical to the server-side validation.

I advocate a combined model based on the intersection between a regular expression and an optional free-form free-form validation algorithm description. Having a regular expression that overlays the free-form algorithm makes it easier to verify the validation logic as a whole because you only need to assess some aspects of the free-form algorightm logic.

In this case we'd specify two loose regular expressions, together with some free-form wording (like what @matsduf suggests above) detailing the additional requirements on top of the regular expressions.

Reply:
I updated this paragraph to mentioned both the free-form wording and regex.

Reply:
I do not think that the regular expressions add anything unless they are updated to correctly capture a valid IP address. The text talks about how it is implemented. That is not part of the API. The same API could be implemented without regular expressions.

Reply:
Can you remove the following lines since they do not make sense in an API document:
```
In addition a preliminary check is performed with the following regexes:
 - IPv4 : `/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/`
 - IPv6 : `/^[0-9a-f:]+(:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})?$/i`
```